### DISCUSSION TITLE
how to apply custom CSS on a per-domain basis?

### DISCUSSION THREAD

**Discussion by uwla** [qutebrowser/qutebrowser#8234] [Q&A]
The `content.user_styles` allows to load custom CSS.

Is it possible to configure qutebrower so that it will load a custom stylesheet based on the current domain?

Here is what I thought but need help:

spawn a user-script which has access to environment variables such as the current domain. This script can then modify `content.user_styles` such that if the domain is `foo.example`, it would load `~/.config/qutebrowser/styles/foo.example.css` by setting `content.user_styles` to `['~/.config/qutebrowser/styles/foo.example.css']`. However, this approach requries manually spawning the user-script. Is there a way to make qutebrowser automatically trigger a user-script on a new page? Writing the script is easy but I don't know how to automatically trigger it on page load.

**Comment by The-Compiler** [comment]
Nope, there isn't. Somewhat related:

- #30 
- #35

Note that your approach wouldn't work nicely though, as toggling the setting will toggle it application-wide; also see:

- #2314

As for your original problem, this PR (which had a few issues that I never got around to look at unfortunately) also has a few workarounds using Greasemonkey support:

- #3854
